src/processors/AvroSchemaInputProcessor.ts

```diff
@@ -0,0 +1,152 @@
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+import {
+  CommonModel,
+  InputMetaModel,
+  AvroSchema,
+  ProcessorOptions
+} from '../models';
+import { Logger } from '../utils';
+import { Interpreter } from '../interpreter/Interpreter';
+import { convertToMetaModel } from '../helpers';
+
+/**
+ * Class for processing Avro Schema input
+ */
+export class AvroSchemaInputProcessor extends AbstractInputProcessor {
+  /**
+   * Function processing an Avro Schema input
+   *
+   * @param input
+   */
+
+  shouldProcess(input?: any): boolean {
+    if (
+      input === '' ||
+      JSON.stringify(input) === '{}' ||
+      JSON.stringify(input) === '[]'
+    ) {
+      return false;
+    }
+    if (!input.type) {
+      return false;
+    }
+    return true;
+  }
+
+  process(input?: any, options?: ProcessorOptions): Promise<InputMetaModel> {
+    if (!this.shouldProcess(input)) {
+      return Promise.reject(
+        new Error('Input is not an Avro Schema, so it cannot be processed.')
+      );
+    }
+    Logger.debug('Processing input as Avro Schema document');
+    const inputModel = new InputMetaModel();
+    inputModel.originalInput = input;
+    input = AvroSchemaInputProcessor.reflectSchemaNames(
+      input,
+      {},
+      'root',
+      true
+    ) as any;
```

Comment:
Not sure if you need to reflect names 🤔 Only necessary if the user does does define all the necessary namings through the standard. I.e. you want to generate an enum, but the avro schema does not have a name for the enum.

Reply:
`reflectSchemaNames` method is adopted to handle the `fields` property in case of `type: record` as follows:

```
{
  "type": "record",
  "name": "Example",
  "doc": "A simple name (attribute) and no namespace attribute: use the null namespace (\"\"); the fullname is 'Example'.",
  "fields": [
    {
      "name": "inheritNull",
      "type": {
        "type": "enum",
        "name": "Simple",
        "doc": "A simple name (attribute) and no namespace attribute: inherit the null namespace of the enclosing type 'Example'. The fullname is 'Simple'.",
        "symbols": ["a", "b"]
      }
    }, {
      "name": "explicitNamespace",
      "type": {
        "type": "fixed",
        "name": "Simple",
        "namespace": "explicit",
        "doc": "A simple name (attribute) and a namespace (attribute); the fullname is 'explicit.Simple' (this is a different type than of the 'inheritNull' field).",
        "size": 12
      }
    }, {
      "name": "fullName",
      "type": {
        "type": "record",
        "name": "a.full.Name",
        "namespace": "ignored",
        "doc": "A name attribute with a fullname, so the namespace attribute is ignored. The fullname is 'a.full.Name', and the namespace is 'a.full'.",
        "fields": [
          {
            "name": "inheritNamespace",
            "type": {
              "type": "enum",
              "name": "Understanding",
              "doc": "A simple name (attribute) and no namespace attribute: inherit the namespace of the enclosing type 'a.full.Name'. The fullname is 'a.full.Understanding'.",
              "symbols": ["d", "e"]
            }
          }
        ]
      }
    }
  ]
}
```

Implemented by observing the `anyOf`, `oneOf` and `allOf` cases of `JSON-Schema`. Couldn't figure out the exact nature of `reflectSchemaName` method.

Reply:
You always have `name` so dont think its necessary to reflect anything 🙂